lib/lambda_ethereum_consensus/store/state_db.ex

```diff
@@ -30,6 +30,45 @@ defmodule LambdaEthereumConsensus.Store.StateDb do
     Db.put(slothash_key_block, block_root)
   end
 
+  def remove_old_states(last_finalized_epoch) do
+    last_finalized_key =
+      (last_finalized_epoch * ChainSpec.get("SLOTS_PER_EPOCH")) |> root_by_slot_key()
+
+    with {:ok, it} <- Db.iterate(),
+         {:ok, @stateslot_prefix <> _slot, _value} <-
+           Exleveldb.iterator_move(it, last_finalized_key),
+         {:ok, slots_to_remove} <- get_slots_to_remove([], it),
+         :ok <- Exleveldb.iterator_close(it) do
+      slots_to_remove |> Enum.map(&remove_by_slot/1)
+    end
+  end
+
+  defp get_slots_to_remove(slots_to_remove, iterator) do
```

Comment:
the naming is a bit confusing. On one hand you are asking for slots to remove but you're also providing it as argument